File: Sieve.cpp
#include <iostream>
#include <vector>

using namespace std;

void sieve_of_eratosthenes(int n) 
{
    vector<bool> prime(n + 1, true);

    for (int p = 2; p * p <= n; p++) 
    {
        if (prime[p] == true) 
        {
            for (int i = p * p; i <= n; i += p) 
            {
                prime[i] = false;
            }
        }
    }

    for (int p = 2; p <= n; p++) 
    {
        if (prime[p] == true) 
        {
            cout << p << " ";
        }
    }
}

int main() 
{
    cout << "Enter number: ";
    int n;
    cin >> n;
    sieve_of_eratosthenes(n);
    return 0;
}

File: Sieve.py
from math import *
ran = int(input('Enter number: '))

numbers = {x: x for x in range(2, ran + 1)}
eres = []
for i in range(2, int(sqrt(ran))+1):
    if numbers[i] == i:
        j = i**2
        while (j < ran + 1):
            numbers[j] = i
            j += i   
for y in range(2, ran + 1):
    if numbers[y] == y:
        eres.append(str(y))
print(','.join(eres))